resources/zib/zib-TreatmentObjective.xml

```diff
@@ -0,0 +1,196 @@
+  <type value="Goal" />
+  <baseDefinition value="http://hl7.org/fhir/StructureDefinition/Goal" />
+  <derivation value="constraint" />
+  <differential>
+    <element id="Goal">
+      <path value="Goal" />
+      <short value="TreatmentObjective" />
+      <comment value="On the zib level, zib TreatmentObjective uses a reference to zib FunctionalOrMentalStatus (in future context). In FHIR the concepts from FunctionalOrMentalStatus are represented directly in the the Goal resource, with the exception of the reference to zib MedicalDevice; this reference is represented by the core extension resource-pertainsToGoal in the [zib-MedicalDevice](http://nictiz.nl/fhir/StructureDefinition/zib-MedicalDevice) profile." />
+      <alias value="Behandeldoel" />
+      <mapping>
+        <identity value="zib-treatmentobjective-v3.2-2020EN" />
+        <map value="NL-CM:13.5.1" />
+        <comment value="TreatmentObjective" />
+      </mapping>
+    </element>
+    <element id="Goal.lifecycleStatus">
+      <path value="Goal.lifecycleStatus" />
+      <comment value="No zib concept can be mapped (implicitly) to this element. Unless `.lifecycleStatus` is explicitly recorded, TreatmentObjectives may assumed to be _active_." />
+    </element>
+    <element id="Goal.description.text">
+      <path value="Goal.description.text" />
+      <short value="DesiredHealthcareResult / StatusName / StatusValue" />
+      <definition value="The textual account of the treatment objective." />
+      <comment value="This element is used to represent the zib concept DesiredHealthcareResult (NL-CM:13.5.5) if present. If instead DesiredHealthCondition::FunctionalOrMentalStatus is used (NL-CM:13.5.3), this element should be populated with a textual representation containg at least FunctionalOrMentalStatus.StatusName (NL-CM:4.26.2) and FunctionalOrMentalStatus.StatusValue (NL-CM:4.26.3).&#xD;&#xA;&#xD;&#xA;Receiving systems that need to map back to the zib could look at `.target.measure` and `.target.detailCodeableConcept`; if these elements are absent, this element represents zib concept DesiredHealthCareResult. If present, the discrete data can be found in the elements mapped to zib FunctionalOrMentalStatus." />
+      <alias value="GewenstZorgresultaat" />
+      <mapping>
+        <identity value="zib-treatmentobjective-v3.2-2020EN" />
+        <map value="NL-CM:13.5.5" />
+        <comment value="DesiredHealthcareResult" />
+      </mapping>
+      <mapping>
+        <identity value="zib-functionalormentalstatus-v3.2-2020EN" />
+        <map value="NL-CM:4.26.2" />
+        <comment value="StatusName (implicit, main mapping is on Goal.target.measure)" />
+      </mapping>
+      <mapping>
+        <identity value="zib-functionalormentalstatus-v3.2-2020EN" />
+        <map value="NL-CM:4.26.3" />
+        <comment value="StatusValue (implicit, main mapping is on Goal.target.detailCodeableConcept)" />
+      </mapping>
+    </element>
+    <element id="Goal.target">

```

Comment:
I'm wondering whether the mapping on this level is correct (instead of a mapping on the root), since:

- when using a FunctionalOrMentalStatus to specify the TreatmentObjective, the FunctionalOrMentalStatus more or less **is** the TreatmentObjective (i.e. the DesiredHealthCondition might fit better on the root of the Goal);
- the FunctionalOrMentalStatus.Comment is part of the DesiredHealthCondition, but is mapped on `Goal.note.text`, which is not part of the `Goal.target`.

Reply:
I think you're right. Too bad, because it was a nice way to mark the container as representing the FunctionalOrMentalStatus.

Moving this mapping also means removing the cardinality constraint on `.target`. I'm not really sure in what situations you'd have multiple targets in a single Goal, but it may be better to lift this restriction.